### ISSUE TITLE
Implementation of New loss function 

### ISSUE THREAD

**Issue by kvmanohar22** [BVLC/caffe#5700]
This involves implementation of the loss function outlined in [this paper](https://arxiv.org/abs/1612.01051).
This loss function is specifically used for the task of object detection. Unlike other object detection algorithms which use complex pipeline where each part must be fine tuned independently and precisely, this regression based loss function can be used to train a model end to end.

**Comment by shaibagon** [comment]
Please use the [caffe-users list][1] for usage, installation, or modeling questions, or other requests for help.
You may also post questions on [stackoverflow][3], make sure you tag them with `caffe` tag.  
There is also [caffe.help][4] documenting the different layers of caffe.
Do not post such requests to Issues. Doing so interferes with the development of Caffe.

Please read the [guidelines for contributing][2] before submitting this issue.

 [1]: https://groups.google.com/forum/#!forum/caffe-users
 [2]: https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md
 [3]: http://stackoverflow.com/questions/tagged/caffe
 [4]: http://caffe.help/manual/layers.html
